### PR TITLE
Android 13 permission related fixes

### LINKED ISSUE #5297: Can't upload pictures after targetSdkVersion bumped to 33
# Steps to reproduce and problems description

1. Reset the app's storage, for instance by uninstalling/reinstalling.
2. Log in.
3. When asked "Allow Commons to access photos and videos on this device?", tap "Allow all".
4. Share a picture to the app.
5. Problem 1: Asked again "Requesting storage permission - We need your permission to access the external storage of your device in order to upload images." I believe I already gave permission at step 2?
6. Problem 2: Background is flickering as seen in the first screencast.
7. Problem 3: After tapping "Settings", it is not clear what needs to be done. See the two screenshots. I am not able to get past the dialog and thus can't upload pictures.

As seen in the second screencast, I also did not find a way to upload using the in-app camera. The custom picker also does not seem usable.

# Version

- This does not occur with 81030d1e78151db3ee436524f8379d6d2dc35a58 ("Fix retried uploads stuck in queued state"), I just tried.
- This starts occurring with the next commit f5770539a5deeb3811082f89b5e15df2c00e7eee ("Target the latest Android version") which means it is currently occurring on the main branch.

I am testing on Android 14.

# Research

https://developer.android.com/about/versions/13/behavior-changes-13#granular-media-permissions
https://developer.android.com/about/versions/12/behavior-changes-12

Interestingly, if I uninstall then install 81030d1e78151db3ee436524f8379d6d2dc35a58 and use it (tap the gallery button then choose picture and reach the caption filling screen) then upgrade to f5770539a5deeb3811082f89b5e15df2c00e7eee without resetting the app's storage, then the problems do not happen (tried just now). This may be related to this API 33 note (and may suggest that getting appropriate READ_MEDIA_* permission(s) would solve this whole issue):

> If your app was previously granted the [READ_EXTERNAL_STORAGE](https://developer.android.com/reference/android/Manifest.permission#READ_EXTERNAL_STORAGE) permission, then any requested READ_MEDIA_* permissions are granted automatically when upgrading.

# Screencast / screenshots

https://github.com/commons-app/apps-android-commons/assets/99590/808341ba-42c3-4274-842e-d1f9ca565e9f

https://github.com/commons-app/apps-android-commons/assets/99590/36d876da-c8e8-4aec-8d8c-0df850916c10

### PR DESCRIPTION
**Pull request by srishti-R**
**Description (required)**

Fixes #5297 

What changes did you make and why?
Updated manifest permissions and runtime storage permissions

**Tests performed (required)**

Tested {build variant, e.g. ProdDebug} on {name of device or emulator} with API level {API level}.
Tested this branch on API 33(android13)

**Screenshots (for UI changes only)**

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -14,6 +14,10 @@
     <uses-permission android:name="android.permission.GET_ACCOUNTS" />
     <uses-permission android:name="android.permission.USE_CREDENTIALS" />
     <uses-permission android:name="android.permission.MANAGE_ACCOUNTS" />
+    <uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
+    <uses-permission android:name="android.permission.READ_MEDIA_VIDEO"/>
+    <uses-permission android:name="android.permission.READ_MEDIA_AUDIO"/>
```
Comment (nicolas-raoul) [inline app/src/main/AndroidManifest.xml:19]: We do not support video nor audio upload for now. Maybe these two permissions could be commented out for now, in order to not ask the user for too many permissions? Or are these actually somehow necessary to access images?

**Review by nicolas-raoul — COMMENTED** [review]
Thanks a lot, it works great in my manual tests! :-)
Unit tests pass too.

File: app/src/main/java/fr/free/nrw/commons/utils/PermissionUtils.java [hunk 2]
```diff
@@ -1,121 +1,146 @@
...
 
 import fr.free.nrw.commons.CommonsApplication;
 import fr.free.nrw.commons.R;
+import java.util.List;
 
 
 public class PermissionUtils {
 
+    public static String[] PERMISSIONS_STORAGE = isSDKVersionScopedStorageCompatible() ?
+        isSDKVersionTiramisu() ? new String[]{Manifest.permission.READ_MEDIA_AUDIO,
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/utils/PermissionUtils.java]: I removed the AUDIO and VIDEO lines here and in the manifest, it seems to work just as well. :-)
Would you mind removing these from the pull request? Or I can do it after merging this pull request, if you don't have time.

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (srishti-R) [reply app/src/main/AndroidManifest.xml]: removed those permissions 👍

File: app/src/main/java/fr/free/nrw/commons/utils/PermissionUtils.java (hunk 2 above)
Comment (srishti-R) [reply app/src/main/java/fr/free/nrw/commons/utils/PermissionUtils.java]: Done, please pull the latest changes in this PR

**Review by nicolas-raoul — APPROVED** [review]
Reinstalled with new changes, it works great.
Thanks Srishti for this brilliant and very fast pull request, just a day after the high priority bug was first reported!

**Comment by Binnette** [comment]
Hi @srishti-R, thank for this fix. I tested latest version from main branch and now I can finally upload pictures from the app with my Android 13 phone. ❤

**Pull request merged**